structure/structure.py

```diff
@@ -1,25 +1,54 @@
+import os
 import sys
 
+from griptape.artifacts import TextArtifact
 from griptape.drivers import (
     GriptapeCloudEventListenerDriver,
 )
 from griptape.events import (
     EventListener,
+    FinishStructureRunEvent,
 )
 from griptape.structures import Agent
 from griptape.tools import Calculator
 
+
+# Returns True if this program is being run from within Griptape Cloud or via the Skatepark emulator
+def is_running_in_managed_environment():
+    return "GT_CLOUD_RUN_ID" in os.environ
+
+
 input = sys.argv[1]
 
+# We need an event driver to communicate events from our program back to Skatepark
```

Comment:
Back to Skatepark _or_ Griptape Cloud Managed Structures

Reply:
so? Those are separate concepts, and we're being explicit.